### PR TITLE
Changed the Asio linkage to fit the THIRDPARTY option and set a minimum version [10322]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: cmake/common/eprosima_libraries.cmake [hunk 1]
```diff
@@ -171,6 +171,9 @@ endmacro()
 #         2.3. Try to find the package again.
 #   3. If the package was not found anywhere, then print an FATAL_ERROR message.
 macro(eprosima_find_thirdparty package thirdparty_name)
+    # Parse arguments.
+    set(oneValueArgs VERSION)
```
Comment (EduPonz) [inline cmake/common/eprosima_libraries.cmake:175]: Please document this argument

File: cmake/modules/FindAsio.cmake [hunk 2]
```diff
@@ -1,19 +1,47 @@
...
+    find_path(ASIO_INCLUDE_DIR NAMES asio.hpp)
+
+    # Not found a local version of Asio
+    if(NOT ASIO_INCLUDE_DIR)
+        # If THIRDPARTY_ASIO=ON the Asio version from thirdparty is used.
+        if(THIRDPARTY_ASIO STREQUAL "ON")
+            find_path(ASIO_INCLUDE_DIR NAMES asio.hpp NO_CMAKE_FIND_ROOT_PATH)
+        # If THIRDPARTY_ASIO=OFF the search is stopped and an error is shown
+        else()
+            message(SEND_ERROR "Not found a local version of Asio installed.")
```
Comment (EduPonz) [inline cmake/modules/FindAsio.cmake:15]: I think these should be `FATAL_ERROR`

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]
LGTM

**Review by EduPonz — APPROVED** [review]
LGTM 🤖

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3569/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/759/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3164/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3601/)

**Review by EduPonz — CHANGES_REQUESTED** [review]
Please correct building errors in Windows

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3573/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/763/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3168/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3605/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3592/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/782/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3182/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3624/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3599/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/789/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3188/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3631/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3609/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/798/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3196/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3640/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3658/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/838/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3222/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3681/)

**Review by imontesino — CHANGES_REQUESTED** [review]

**Review by imontesino — APPROVED** [review]
LGTM

**Pull request merged**